src/sources/soundsourceflac.cpp

```diff
@@ -403,12 +403,18 @@ namespace {
 // get rid of the garbage in the most significant bits before scaling
 // to the range [-CSAMPLE_PEAK, CSAMPLE_PEAK - epsilon] with
 // epsilon = 1 / 2 ^ bitsPerSample.
-constexpr CSAMPLE kSampleScaleFactor = CSAMPLE_PEAK /
+//
+// We have to negate the nominator to compensate for the negative denominator!
+// Otherwise the phase would be inverted: https://bugs.launchpad.net/mixxx/+bug/1933001
+constexpr CSAMPLE kSampleUnshiftFactor = -CSAMPLE_PEAK /
         (static_cast<FLAC__int32>(1) << std::numeric_limits<FLAC__int32>::digits);
 
 inline CSAMPLE convertDecodedSample(FLAC__int32 decodedSample, int bitsPerSample) {
-    DEBUG_ASSERT(std::numeric_limits<FLAC__int32>::is_signed);
-    return (decodedSample << ((std::numeric_limits<FLAC__int32>::digits + 1) - bitsPerSample)) * kSampleScaleFactor;
+    static_assert(std::numeric_limits<FLAC__int32>::is_signed);
+    static_assert(kSampleUnshiftFactor > CSAMPLE_ZERO, "preserve phase of decoded signal");
+    const auto shiftLeft = (std::numeric_limits<FLAC__int32>::digits + 1) - bitsPerSample;
+    const auto shiftedSample = decodedSample << shiftLeft;
+    return shiftedSample * kSampleUnshiftFactor;
```

Comment:
I think this deserves a test. 

Lets assume 8 bitsPerSample where the minimum is  -128 (0x000000FF)
shiftLeft = 31 + 1 - 8 = 24
shiftedSample = decodedSample << shiftLeft
-128 -> 0x0xFF000000
kSampleUnshiftFactor = 1 / 0x8000000
0xFF000000 / 0x8000000 != -1.0f 

Test Failed :-(

```suggestion
    return -CSAMPLE_PEAK * shiftedSample / (static_cast<FLAC__int32>(-1) << shiftLeft); 
```

Reply:
Indeed, I have used a confusing naming for the constant and temporary variables. Fixed.

Your example is wrong: 0x000000FF = -1, 0x00000080 = -128